tests/testsuite/build.rs

```diff
@@ -5972,6 +5972,62 @@ fn build_with_relative_cargo_home_path() {
     p.cargo("build").env("CARGO_HOME", "./cargo_home/").run();
 }
 
+#[cargo_test]
+fn skip_mtime_check_in_selected_cargo_home_subdirs() {
+    // Run once without cache
+    let p = project()
+        .at("cargo_home/registry/foo")
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/main.rs", r#"fn main() { println!("1"); }"#)
+        .build();
+    let project_root = p.root();
+    let cargo_home = project_root
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .to_str()
+        .unwrap();
+    p.cargo("run")
+        .env("CARGO_HOME", &cargo_home)
+        .with_stdout("1")
+        .run();
+    // Run with cache
+    p.change_file("src/main.rs", r#"fn main() { println!("2"); }"#);
+    p.cargo("run")
+        .env("CARGO_HOME", &cargo_home)
+        .with_stdout("1")
+        .run();
+}
+
+#[cargo_test]
+fn use_mtime_cache_in_cargo_home() {
+    println!("Without cache");
+    let p = project()
+        .at("cargo_home/foo")
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/main.rs", r#"fn main() { println!("1"); }"#)
+        .build();
+    let project_root = p.root();
+    let cargo_home = project_root
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .to_str()
+        .unwrap();
```

Comment:
Should be only one level up, right?

```suggestion
        .unwrap();
```